underfs/swift/src/main/java/tachyon/underfs/swift/direct/SwiftDirectUnderFileSystem.java

```diff
@@ -0,0 +1,400 @@
+    LOG.warn("Create with block size and replication is not"
+        + "supported with SwiftDirectUnderFileSystem."
+        + " Block size and replication will be ignored.");
+    return create(path);
+  }
+
+  /* (non-Javadoc)
+   * @see tachyon.underfs.UnderFileSystem#delete(java.lang.String, boolean)
+   *
+   * recursive parameter is not being used for delete.
+   * The reason is that each path is treated as a single object, which
+   * was created with 'create' method.
+   *
+   * Method return 'true' always.
+   */
+  @Override
+  public boolean delete(String path, boolean recursive) throws IOException {
+    LOG.debug("Delete method: " + path + ",recursive: " + recursive);
+    String strippedPath = stripPrefixIfPresent(path);
+    LOG.debug("Going to retrieve container: " + mContainerName);
+    Container c = mAccount.getContainer(mContainerName);
+    LOG.debug("Going to retrieve an object: " + strippedPath);
+    StoredObject so = c.getObject(strippedPath);
+    LOG.debug("Checking if object exists in Swift. If so, delete it");
+    if (so.exists()) {
+      LOG.debug("Object exists: " + so.getBareName());
+      so.delete();
+    }
+    return true;
+  }
+
+  @Override
+  public boolean exists(String path) throws IOException {
+    LOG.debug("Exists: " + path);
+    String newPath = stripPrefixIfPresent(path);
+    return getObjectDetails(newPath) ;
+  }
+
+  /**
+   * Gets the StorageObject representing the metadata of a key. If the key does not exist as a
+   * file or folder, null is returned
+   * @param key the key to get the object details of
+   * @return StorageObject of the key, or null if the key does not exist as a file or folder
+   */
+  private boolean getObjectDetails(String path) {
+    LOG.debug("Get object details: " + path);
+    boolean res =  mAccount.getContainer(mContainerName).getObject(path).exists();
+    LOG.debug("Result: " + res);
+    return res;
+  }
+
+  /**
+   * There is no concept of a block in Swift, however the maximum allowed size of
+   * one object is currently 4 GB.
+   *
+   * @param path to the file name
+   * @return 4 GB in bytes
+   * @throws IOException
+   */
+  @Override
+  public long getBlockSizeByte(String path) throws IOException {
+    return Constants.GB * 4;
+  }
+
+  @Override
+  public Object getConf() {
+    LOG.warn("getConf is not supported when using SwiftDirectUnderFileSystem, returning null.");
+    return null;
+  }
+
+  @Override
+  public List<String> getFileLocations(String path) throws IOException {
+    LOG.warn("getFileLocations is not supported when using "
+        + "SwiftDirectUnderFileSystem, returning null.");
+    return null;
+  }
+
+  @Override
+  public List<String> getFileLocations(String path, long offset) throws IOException {
+    LOG.warn("getFileLocations is not supported when using "
+        + "SwiftDirectUnderFileSystem, returning null.");
+    return null;
+  }
+
+  @Override
+  public long getFileSize(String path) throws IOException {
```

Comment:
what if `path` doesn't exist? shouldn't we throw `FileNotFoundException(path);`like in `S3UnderFileSystem`